### PR TITLE
gcp: Add most install components

### PR DESCRIPTION
**Pull request by nbrahms**
This commit adds all GCP install components except for the org-wide-policy component. This will come in a later PR.

This work uses the new root metadata on the GCP root install, as well as item-specific metadata.

Components are structured as follows:

- p0_gcp: The root organization install; holds metadata for the access-logs component
- p0_access_logs: Installable without staging, as proof of project ownership is determined by having a currently installed iam-assessment component
- p0_iam_assessment: Requires a staging install to prove ownership
- p0_iam_write: Requires a staging install to prove ownership
- p0_sharing_restriction: Installable without staging as it depends on an installed iam-write component

For each component, I added detailed instructions on peer Terraform that must be installed alongside.

### CODE REVIEW

File: internal/provider/resources/install/gcp/sharing_restriction.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	project = p0_gcp_iam_write_staged.example.project
+}
+
+# Prevents P0 from assigning grants outside your domain
+resource "google_org_policy_policy" "example" {
+  name   = "projects/my_project/policies/iam.allowedPolicyMemberDomains"
+  parent = "projects/my_project"
+
+  spec {
+    rules {
```
Comment (GGonryun) [inline internal/provider/resources/install/gcp/sharing_restriction.go:60]: Nit: the spacing is funny here, but the generated content looks file 🤔.

File: examples/resources/p0_gcp_iam_write/resource.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# The predefined role is required for P0 to grant resource-level access in your project
+resource "google_project_iam_member" "example_predefined_role" {
+  project = locals.project
+  role    = p0_gcp_iam_write_staged.example.predefined_role
+  member  = "serviceAccount:${p0_gcp.example.service_account_email}"
+}
+
+# The `p0_gcp_iam_write` resource will fail to validate unless it is installed
+# _after_ the P0 service account is granted the above roles
+resource "p0_gcp_iam_write" "example" {
```
Comment (GGonryun) [inline examples/resources/p0_gcp_iam_write/resource.tf:38]: Does the `example` resource name need to be unique to run these samples?

**Review by GGonryun — APPROVED** [review]
LGTM, a few small comments and it would be nice to have examples for the other resources like `access_logs`, `iam_assessment`, and `sharing_restriction`.

File: examples/resources/p0_gcp_iam_write/resource.tf (hunk 2 above)
Comment (nbrahms) [reply examples/resources/p0_gcp_iam_write/resource.tf:38]: no, they're namespaced by the resource type

**Pull request merged**